internal/core/autocomplete.go

```diff
@@ -116,11 +173,27 @@ func AutoComplete(ctx context.Context, leftWords []string, wordToComplete string
 
 	// For each left word that is not a flag nor an argument, we try to go deeper in the autocomplete tree and store the current node in `node`.
 	node := commandTreeRoot
-	for i, word := range leftWords {
+	for i := 0; i < len(leftWords); i++ {
+		word := leftWords[i]
 		children, exist := node.Children[word]
-		if i < len(leftWords) && exist && !children.IsArg {
+
+		switch {
+		case !exist && node.isLeafCommand():
+			// word is probably an unknown argument
+			// Just skip it
+
+		case !exist:
+			// we did not reach a leaf command, and word is unknown
+			return &AutocompleteResponse{}
+
+		case children.Type == AutoCompleteNodeTypeArgument:
```

Comment:
Add a comment explaining why you do nothing for an argument (but continue to dig for flag)